### PR TITLE
Convert Repo Scripts To Published TypeScript Packages

### PR DESCRIPTION
**Pull request by NickGerleman**
This change was motivated by a desire to update release note generation formatting and logic without the need to backport scripts through different versions of RNW. Instead of backporting, we can have the publish pipeline for stable branches pull the latest published script.

These scripts have become essential enough where the move to Typescript makes sense, and having a single package sharing dependencies between them ends up muddling the waters. Scripts are not always self-isolated, like the shared find-repo-root script. Because public scripts may have dependencies, I marked all scripts scoped to "@rnw-scripts" to be published to avoid scenarios of accidentally introducing missing dependencies.

Some of this adds overhead/boilerplate for simple one-off scripts, but overall sets a more maintainable pattern going forward.

For servicing release note generation changes, we use the release promotion script to use the local release of create-github-releases in the master branch and switch to the latest published version in stable branches.

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/react-native-windows/pull/5352)

### CODE REVIEW

File: packages/@rnw-scripts/create-github-releases/src/createGithubReleases.ts [hunk 1]
```diff
@@ -9,27 +9,81 @@
...
+  entries: Array<{
+    date: string;
+    tag: string;
+    version: string;
+    comments: {[type in 'none' | 'prerelease' | 'minor' | 'major']: Comment[]};
+  }>;
+}
+
+/**
+ * Represents the details of an individual relaease
```
Comment (NickGerleman) [inline packages/@rnw-scripts/create-github-releases/src/createGithubReleases.ts:53]: Nit typo

File: packages/@rnw-scripts/create-github-releases/src/createGithubReleases.ts [hunk 2]
```diff
@@ -9,27 +9,81 @@
...
+ */
+interface Release {
+  packageName: string;
+  tag: string;
+  version: string;
+  comments: Array<{
```
Comment (NickGerleman) [inline packages/@rnw-scripts/create-github-releases/src/createGithubReleases.ts:59]: This should just be Comment[]

File: lerna.json [hunk 3]
```diff
@@ -1,5 +1,5 @@
 {
-  "packages": ["packages/*", "vnext"],
+  "packages": ["packages/**", "vnext"],
```
Comment (acoates-ms) [inline lerna.json:2]: Not sure how much is really matters, (how lerna does its file trolling?), but maybe packages/* and packages/@rnw-scripts/* would save it some effort?

File: package.json [hunk 4]
```diff
@@ -22,7 +22,7 @@
   "license": "MIT",
   "workspaces": {
     "packages": [
-      "packages/*",
+      "packages/**",
```
Comment (acoates-ms) [inline package.json:25]: Same comment as above

File: packages/@rnw-scripts/create-github-releases/tsconfig.json [hunk 5]
```diff
@@ -0,0 +1,16 @@
+{
```
Comment (dannyvv) [inline packages/@rnw-scripts/create-github-releases/tsconfig.json:1]: NIT: Nothing you can do, just sharing... :)
I wished yarn workspaces or lerna could deal with 'default' configs for packages... How many copies of just-task and tsconfig we have in our repo is getting a bit silly...

File: packages/@rnw-scripts/promote-release/src/promoteRelease.ts [hunk 6]
```diff
@@ -203,10 +208,10 @@ async function updatePackage(packageName, props) {
 /**
  * Get the npm distribution tag for a given version
  *
- * @param {string} release
- * @param {string} version
+ * @param release
+ * @param version
  */
-function distTag(release, version) {
+function distTag(release: ReleaseType, version: string) {
```
Comment (NickGerleman) [inline packages/@rnw-scripts/promote-release/src/promoteRelease.ts:214]: Add return type

File: packages/@rnw-scripts/create-github-releases/src/createGithubReleases.ts [hunk 7]
```diff
@@ -204,10 +246,8 @@ function createReleaseMarkdown(release) {
 
 /**
  * Create request headers common to GitHub API calls
- *
- * @param {string} token
  */
-function requestHeaders(token) {
+function requestHeaders(token: string) {
```
Comment (NickGerleman) [inline packages/@rnw-scripts/create-github-releases/src/createGithubReleases.ts:250]: Add return type

File: packages/@rnw-scripts/create-github-releases/src/createGithubReleases.ts [hunk 8]
```diff
@@ -217,10 +257,8 @@ function requestHeaders(token) {
 
 /**
  * Return a release title name corresponding to a package
- *
- * @param {string} packageName
  */
-function packageTitle(packageName) {
+function packageTitle(packageName: string) {
```
Comment (NickGerleman) [inline packages/@rnw-scripts/create-github-releases/src/createGithubReleases.ts:261]: Add return type

File: packages/@rnw-scripts/create-github-releases/src/createGithubReleases.ts [hunk 9]
```diff
@@ -163,35 +209,31 @@ async function publishRelease(release, token) {
...
   }));
 }
 
 /**
  * Create the markdown representation of a release
- *
- * @param {object} changeEntry
  */
-function createReleaseMarkdown(release) {
+function createReleaseMarkdown(release: Release) {
```
Comment (NickGerleman) [inline packages/@rnw-scripts/create-github-releases/src/createGithubReleases.ts:236]: Add return type

File: lerna.json (hunk 3 above)
Comment (NickGerleman) [reply lerna.json:2]: Running glob.sync for packages/** as a test of the worst case was really slow. I'll make that change.

File: packages/@rnw-scripts/create-github-releases/tsconfig.json (hunk 5 above)
Comment (NickGerleman) [reply packages/@rnw-scripts/create-github-releases/tsconfig.json:1]: We could potentially have a build script to populate them. I think we've done something similar before actually. I'm not sure it's worth the complexity yet, but could be a consideration if we see any maintenance headaches with these.

**Pull request merged**